### PR TITLE
pico-imx7d: Enable loading kernel from rootfs partition

### PR DESCRIPTION
**Pull request by vanmaegima**
Replace fatload support for kernel loading with the fs generic loading interface
and enable booting from a single partition to support booting with a wic image.

These are needed since the wic scripts available do not generate a Boot (FAT)
partition.

Signed-off-by: Vanessa Maegima <vanessa.maegima@nxp.com>

### CODE REVIEW

File: include/configs/pico-imx7d.h [hunk 1]
```diff
@@ -38,20 +38,20 @@
...
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
-	"finduuid=part uuid mmc 0:2 uuid\0" \
+	"finduuid=part uuid mmc 0:1 uuid\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
 		"root=PARTUUID=${uuid} rootwait rw\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadimage=ext4load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
+	"loadfdt=ext4load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
```
Comment (otavio) [inline include/configs/pico-imx7d.h]: This can be reworked and send upstream. Use 'load mmc' so it uses the fs generic interface and is agnostic.

After this commit, make another commit that adapts the code to the single partition use.

**Review by otavio — CHANGES_REQUESTED** [review]

**Comment by vanmaegima** [comment]
Pull request updated.
